tgui/packages/tgui/interfaces/ListInputModal.tsx

```diff
@@ -0,0 +1,193 @@
+import { Loader } from './common/Loader';
+import { InputButtons, Preferences, Validator } from './common/InputButtons';
+import { Button, Input, Section, Stack } from '../components';
+import { KEY_ENTER, KEY_DOWN, KEY_UP } from 'common/keycodes';
+import { Window } from '../layouts';
+import { useBackend, useSharedState } from '../backend';
+
+type ListInputData = {
+  items: string[];
+  message: string;
+  preferences: Preferences;
+  timeout: number;
+  title: string;
+};
+
+export const ListInputModal = (_, context) => {
+  const { act, data } = useBackend<ListInputData>(context);
+  const { items = [], message, preferences, title, timeout } = data;
+  const { large_buttons } = preferences;
+  const [selected, setSelected] = useSharedState<number | null>(
+    context,
+    'input',
+    0
+  );
+  const [searchBarVisible, setSearchBarVisible] = useSharedState<boolean>(
+    context,
+    'searchBarVisible',
+    items.length > 9
+  );
+  const [searchQuery, setSearchQuery] = useSharedState<string>(
+    context,
+    'searchQuery',
+    ''
+  );
+  const [inputIsValid, setInputIsValid] = useSharedState<Validator>(
+    context,
+    'inputIsValid',
+    { isValid: true, error: null }
+  );
```

Comment:
None of these should be `useSharedState` (I'm hesitant to say anything should be), use local state.